components/public-api/typescript/src/metrics.ts

```diff
@@ -237,31 +238,40 @@ export function getMetricsInterceptor(): Interceptor {
         } finally {
             if (settled) {
                 stopTimer({ grpc_code: status ? Code[status] : "OK" });
-                GRPCMetrics.handled({ ...labels, code: status ? Code[status] : "OK" });
+                metrics.handled({ ...labels, code: status ? Code[status] : "OK" });
             }
         }
     };
 }
 
+export type MetricsRequest = RequestInit & { url: string };
+
 export class MetricsReporter {
     private static readonly REPORT_INTERVAL = 10000;
 
     private intervalHandler: NodeJS.Timer | undefined;
 
     private readonly metricsHost: string;
 
+    private readonly pendingRequests: MetricsRequest[] = [];
```

Comment:
we could consider preserving them in local storage, but it makes things complex, so won't do it for now

Reply:
We can add them now probably, I'm just concernted that localstorage is small and it can grow, maybe we put it in indexed db, not sure.

Annother problem is a race we should make it sure that between all windows it is reported only once.